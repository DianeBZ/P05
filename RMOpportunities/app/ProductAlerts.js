import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  TextInput,
  TouchableHighlight,
} from 'react-native';

import Translation from './Translation';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

export default class ProductAlerts extends Component{
  constructor(props) {
    super(props);
    this.state = {text: ''};
  }
    search= () =>{
      alert(this.state.text+' a été bien ajouté');
    }
  render(){
    return(
      <View style={styles.container}>
        <View style={{height:height*0.08}}/>
        <View style={styles.containerMain}>
          <View style = {styles.containerText}>
            <Text style = {styles.title}>
				{Translation[lang].al_prod}
            </Text>
            <Text style = {styles.instructions}>
			  {Translation[lang].al_prod_text}
            </Text>
          </View>
          <View style={styles.containerAjouteAlerte}>
            <TextInput style={styles.textToFill} underlineColorAndroid={'transparent'}
            placeholder={Translation[lang].num_id}
            onChangeText={(text) => this.setState({text})}
            />
            <TouchableHighlight style={styles.bouton}>
              <Text style={{fontSize:height*0.03,color:'white'}} onPress={this.search.bind(this)}>
				  {Translation[lang].aj}
              </Text>
            </TouchableHighlight>
          </View>
          <View style = {{justifyContent: 'flex-start',alignSelf: 'flex-start',marginLeft:5}}>
            <Text style = {styles.titleAV}>
                {Translation[lang].al_def}
            </Text>
          </View>
          <View style = {styles.header}>
            <View style = {styles.listHeaderFirst}>
              <Text style = {{textAlign: 'center',fontSize:10,}}>
                {Translation[lang].num_id}
              </Text>
            </View>
            <View style = {styles.listHeader}>
              <Text style = {{textAlign: 'center',fontSize:10,}}>
				  {Translation[lang].activer}
              </Text>
            </View>
            <View style = {styles.listHeader}>
              <Text style = {{textAlign: 'center',fontSize:10,}}>
				  {Translation[lang].supprimer}
              </Text>
            </View>
          </View>
          <View style={styles.containerContenu}>

          </View>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  containerMain: {
    backgroundColor: '#F5FCFF',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.89,
	alignSelf: 'center',
  },
  containerText: {
    backgroundColor: '#F5FCFF',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.28,
	alignSelf: 'center',
  },
  title: {
    fontSize: 40,
    textAlign: 'center',
    margin: 10,
	  color: '#333333',
  },
  instructions: {
    fontSize: 15,
	   textAlign: 'center',
    marginBottom:5,
  },
  containerAjouteAlerte:{
    margin:10,
    flexDirection:'row',
  },
  textToFill:{
    width:width*0.6,
    height:height*0.05,
    fontSize:11,
    padding:5,
    borderWidth:1,
    borderColor:'#CCC',
  },
  bouton:{
    width:width*0.2,
    height:height*0.05,
    backgroundColor:'#A4D04A',
    alignItems:'center',
    justifyContent:'center',
  },
  titleAV: {
  fontSize: 20,
  color: '#333333',
  textAlign: 'left',
  },
  header: {
  width: width,
  //height: height * 0.12,
  //alignItems: 'flex-end',
  flexDirection: 'row',
  justifyContent: 'center',
  },
  listHeader: {
  justifyContent: 'center',
  alignSelf: 'center',
    padding: 5,
    width: 0.16 * width,
    height: height * 0.05,
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: '#CCC',

  },
  listHeaderFirst:{
    justifyContent: 'center',
    alignSelf: 'center',
    padding: 5,
    width: 0.4 * width,
    height: height * 0.05,
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: '#CCC',
  },
  containerContenu: {
  width: width*0.72,
  flexDirection:'row',
  height: height * 0.23,
  backgroundColor: '#F5FCFF',
  padding: 5,
  borderWidth: 1,
  borderRadius: 5,
  margin: 5,
  borderColor: '#CCC',
  },
});
