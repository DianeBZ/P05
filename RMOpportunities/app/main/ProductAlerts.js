import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  TextInput,
  TouchableHighlight,
  BackAndroid,
} from 'react-native';

import Translation from './Translation';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

export default class ProductAlerts extends Component{
  constructor(props) {
    super(props);
    this.state = {text: ''};
  }// 'text' is the content(registry number) inputed by the user
    search= () =>{
      alert(this.state.text+ Translation[lang].bien_ajout√©);
    }// inform the user that the alert is added successfully
  render(){
	BackAndroid.addEventListener('Back',this.onBackAndroid);
    return(
      <View style={styles.container}>
        <View style={{height:height*0.08}}/>
        <View style={styles.containerMain}>
          <View style = {styles.containerText}>
            <Text style = {styles.title}>
				{Translation[lang].al_prod}
            </Text>
            <Text style = {styles.instructions}>
			  {Translation[lang].al_prod_text}
            </Text>
          </View>
          <View style={styles.containerAjouteAlerte}>
            <TextInput style={styles.textToFill} underlineColorAndroid={'transparent'}
            placeholder={Translation[lang].num_id}
            onChangeText={(text) => this.setState({text})}
            />
            <TouchableHighlight style={styles.bouton}>
              <Text style={{fontSize:height*0.03,color:'white'}} onPress={this.search.bind(this)}>
				  {Translation[lang].aj}
              </Text>
            </TouchableHighlight>
          </View>
          <View style = {{justifyContent: 'flex-start',alignSelf: 'flex-start',marginLeft:5}}>
            <Text style = {styles.titleAV}>
                {Translation[lang].al_def}
            </Text>
          </View>
          <View style = {styles.header}>
            <View style = {styles.listHeaderFirst}>
              <Text style = {{textAlign: 'center',fontSize:10,}}>
                {Translation[lang].num_id}
              </Text>
            </View>
            <View style = {styles.listHeader}>
              <Text style = {{textAlign: 'center',fontSize:10,}}>
				  {Translation[lang].activer}
              </Text>
            </View>
            <View style = {styles.listHeader}>
              <Text style = {{textAlign: 'center',fontSize:10,}}>
				  {Translation[lang].supprimer}
              </Text>
            </View>
          </View>
          <View style={styles.containerContenu}>

          </View>
        </View>
      </View>
    )
  }
      onBackAndroid = () => {
		const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F2F2F2',
  },
  containerMain: {
    backgroundColor: '#F2F2F2',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.89,
	alignSelf: 'center',
  },
  containerText: {
    backgroundColor: '#F2F2F2',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.28,
	alignSelf: 'center',
  },
  title: {
    fontSize: 40,
    textAlign: 'center',
    margin: 10,
	  color: '#333333',
  },
  instructions: {
    fontSize: 15,
	   textAlign: 'center',
    marginBottom:5,
  },
  containerAjouteAlerte:{
    margin:10,
    flexDirection:'row',
  },
  textToFill:{
    width:width*0.6,
    height:height*0.05,
    fontSize:11,
    padding:5,
    borderWidth:1,
    borderColor:'#CCC',
  },
  bouton:{
    width:width*0.2,
    height:height*0.05,
    backgroundColor:'#A4D04A',
    alignItems:'center',
    justifyContent:'center',
  },
  titleAV: {
  fontSize: 20,
  color: '#333333',
  textAlign: 'left',
  },
  header: {
  width: width,
  //height: height * 0.12,
  //alignItems: 'flex-end',
  flexDirection: 'row',
  justifyContent: 'center',
  },
  listHeader: {
  justifyContent: 'center',
  alignSelf: 'center',
    padding: 5,
    width: 0.16 * width,
    height: height * 0.05,
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: '#CCC',

  },
  listHeaderFirst:{
    justifyContent: 'center',
    alignSelf: 'center',
    padding: 5,
    width: 0.4 * width,
    height: height * 0.05,
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: '#CCC',
  },
  containerContenu: {
  width: width*0.72,
  flexDirection:'row',
  height: height * 0.23,
  backgroundColor: '#F2F2F2',
  padding: 5,
  borderWidth: 1,
  borderRadius: 5,
  margin: 5,
  borderColor: '#CCC',
  },
});
