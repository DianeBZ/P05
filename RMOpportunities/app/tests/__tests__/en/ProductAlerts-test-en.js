// __tests__/ProductAlerts-test-en.js
import 'react-native';
import React from 'react';

global.lang = 'en';
import ProductAlerts from '../../../main/ProductAlerts';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <ProductAlerts />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
