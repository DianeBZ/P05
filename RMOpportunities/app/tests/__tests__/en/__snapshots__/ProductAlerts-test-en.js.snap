// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "height": 106.72,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#F2F2F2",
        "flexDirection": "column",
        "flexGrow": 0,
        "flexShrink": 0,
        "height": 1187.26,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F2F2F2",
          "flexDirection": "column",
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 373.52000000000004,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 40,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Product alerts
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
            "marginBottom": 5,
            "textAlign": "center",
          }
        }
      >
        Add an alert on a product in order to be notified when it appears on the plateform
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "margin": 10,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Registry number (CAS or EC)"
        style={
          Object {
            "borderColor": "#CCC",
            "borderWidth": 1,
            "fontSize": 11,
            "height": 66.7,
            "padding": 5,
            "width": 450,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#A4D04A",
              "height": 66.7,
              "justifyContent": "center",
              "width": 150,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 40.019999999999996,
            }
          }
        >
          Add
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "justifyContent": "flex-start",
          "marginLeft": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 20,
            "textAlign": "left",
          }
        }
      >
        Existing alerts
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "borderColor": "#CCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 66.7,
            "justifyContent": "center",
            "padding": 5,
            "width": 300,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "textAlign": "center",
            }
          }
        >
          Registry number (CAS or EC)
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "borderColor": "#CCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 66.7,
            "justifyContent": "center",
            "padding": 5,
            "width": 120,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "textAlign": "center",
            }
          }
        >
          Activate
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "borderColor": "#CCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 66.7,
            "justifyContent": "center",
            "padding": 5,
            "width": 120,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "textAlign": "center",
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "borderColor": "#CCC",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 306.82,
          "margin": 5,
          "padding": 5,
          "width": 540,
        }
      }
    />
  </View>
</View>
`;
